config_ext_test.go

```diff
@@ -447,57 +418,210 @@ mismatch: [scalar]
+		c := struct {
+			Toyota string
+			Honda  string
+			Nissan string `yaml:",flow"`
+		}{}
+		require.NoError(t, provider.Get("practical").Populate(&c))
+		assert.Equal(t, "altima", c.Nissan, "wrong Nissan")
+	})
 
-		var c cfg
-		require.NoError(t, provider.Get(Root).Populate(&c))
-		assert.Empty(t, c.Boring, "wrong number of fun cars")
+	t.Run("inline field", func(t *testing.T) {
+		t.Skip("TODO: doesn't handle inline field")
+		c := struct {
+			Humans struct {
+				Driver    string
+				Passenger string
+				Backseat  []string
+			} `yaml:",inline"`
+		}{}
+		require.NoError(t, provider.Get("occupants.honda").Populate(&c))
+		assert.Equal(t, 1, len(c.Humans.Backseat), "wrong number of backseat occupants")
 	})
+}
 
-	t.Run("omitempty field", func(t *testing.T) {
-		t.Skip("TODO: doesn't handle omitempty fields")
-		type cfg struct {
-			Fun []string `yaml:",omitempty"`
-		}
+func TestPermissiveYAML(t *testing.T) {
+	provider, err := NewYAMLProviderFromReader(
+		strings.NewReader(_base),
+		strings.NewReader(_override),
+	)
+	require.NoError(t, err, "couldn't create provider")
+	assert.Equal(t, `cached "yaml"`, provider.Get(Root).Source(), "wrong source")
+	assert.Equal(t, "<nil>", provider.Get("nothing").String(), "wrong string representation")
 
-		var c cfg
-		require.NoError(t, provider.Get(Root).Populate(&c))
-		assert.Equal(t, 2, len(c.Fun), "wrong number of fun cars")
+	runCommonTests(t, provider)
+
+	t.Run("ignore type mismatch during merge", func(t *testing.T) {
+		provider, err := NewYAMLProviderFromReader(
+			strings.NewReader("mismatch: foo"),   // scalar
+			strings.NewReader("mismatch: [foo]"), // sequence
+		)
+		require.NoError(t, err, "couldn't create permissive provider with type mismatch")
+
+		var s []string
+		run(t, testCase{
+			Value:     provider.Get("mismatch"),
+			HasValue:  true,
+			Interface: []interface{}{"foo"},
+			Populate:  &s,
+		})
+		assert.Equal(t, []string{"foo"}, s, "unexpected result after populate")
 	})
 
-	t.Run("flow field", func(t *testing.T) {
-		t.Skip("TODO: doesn't handle flow fields")
-		type cfg struct {
-			Fun []string `yaml:",flow"`
-		}
+	t.Run("ignores type mismatches during WithDefault", func(t *testing.T) {
+		// Since defaults are effectively the lowest-priority config, their types
+		// should be ignored when necessary.
+		provider, err := NewYAMLProviderFromReader(
+			strings.NewReader("mismatch: foo"), // scalar
+		)
+		require.NoError(t, err, "couldn't create provider")
+		val, err := provider.Get("mismatch").WithDefault([]string{"foo"}) // sequence
+		assert.NoError(t, err, "error on type mismatch in permissive mode")
+		assert.Equal(t, "foo", val.Value(), "wrong merged output")
+	})
 
-		var c cfg
-		require.NoError(t, provider.Get(Root).Populate(&c))
-		assert.Equal(t, 2, len(c.Fun), "wrong number of fun cars")
+	t.Run("ignore duplicate keys", func(t *testing.T) {
+		provider, err := NewYAMLProviderFromReader(
+			strings.NewReader("dupe: foo\ndupe: bar"),
```

Comment:
The last configured wins?

Reply:
Yep. That's what the YAML specification calls for :man_shrugging: